distributed/scheduler.py

```diff
@@ -6778,22 +6776,14 @@ def _task_to_report_msg(state: SchedulerState, ts: TaskState) -> dict:
 @cfunc
 @exceptval(check=False)
 def _task_to_client_msgs(state: SchedulerState, ts: TaskState) -> dict:
-    cs: ClientState
-    client_keys: list
+    report_msg: dict = _task_to_report_msg(state, ts)
     if ts is None:
         # Notify all clients
-        client_keys = list(state._clients)
+        return {k: [report_msg] for k in state._clients}
```

Comment:
Should this be first typed as dict rather than building on the fly as part of the return .  I am seeing that technique used for `recommendations` above

Reply:
We can go either way. Did it on the fly here as it kept the code succinct. Open to changing it though

With `recommendations` am trying to line it up with how we handle other return arguments (like `client_msgs` and `worker_msgs`), which are also assigned up front. This also avoids accidental duplication (like assigning `recommendations` with `{}` in a few places). Thus allowing us to pay the cost of creating the `dict` only once and then using that `dict` or modifying it later

Reply:
Thanks for the explanation.   I don't have a strong opinion here and would defer to you.  I brought it up naively because of the techniques I saw earlier

Reply:
Makes sense. Went ahead and assigned it to a variable for consistency